### PR TITLE
ENHANCEMENT: Better communication of $owns behaviour

### PR DESCRIPTION
**Pull request by unclecheese**
Resolves https://github.com/silverstripe/silverstripe-versioned/issues/71

### CODE REVIEW

**Comment by flamerohr** [comment]
Needs https://github.com/silverstripe/silverstripe-admin/pull/418/files merged

Screenshot:
screen shot 2018-02-02 at 10 38 12 am

File: src/VersionedGridFieldDetailForm.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ public function updateItemRequestClass(&$class, $gridField, $record, $requestHan
     {
         // Conditionally use a versioned item handler if it doesn't already have one.
         if ($record
-            && $record->has_extension(Versioned::class)
+            && ($record->has_extension(Versioned::class) || $record->hasExtension(RecursivePublishable::class))
```
Comment (flamerohr) [inline src/VersionedGridFieldDetailForm.php]: shifted this to variables - also fails linting

File: src/VersionedGridFieldItemRequest.php [hunk 2]
```diff
@@ -42,69 +43,39 @@ public function Breadcrumbs($unlinked = false)
...
+    /**
+     * Ensures that an unversioned object calls publishRecursive() to its ownees
+     * @param array $data
+     * @param Form $form
+     */
+    public function saveFormIntoRecord($data, $form)
+    {
+        $record = parent::saveFormIntoRecord($data, $form);
+        $record->publishRecursive();
+    }
```
Comment (flamerohr) [inline src/VersionedGridFieldItemRequest.php]: missing `return $record;`

File: src/VersionedGridFieldItemRequest.php [hunk 3]
```diff
@@ -257,4 +228,91 @@ protected function getRecordStatus()
 
         return null;
     }
+
+    protected function addVersionedButtons(DataObject $record, FieldList $actions)
```
Comment (flamerohr) [inline src/VersionedGridFieldItemRequest.php]: missing docblock

File: src/VersionedGridFieldItemRequest.php [hunk 4]
```diff
@@ -257,4 +228,91 @@ protected function getRecordStatus()
...
+                    ->setDescription(_t(
+                        __CLASS__ . '.BUTTONARCHIVEDESC',
+                        'Unpublish and send to archive'
+                    ))
+                    ->addExtraClass('delete btn-secondary')
+            );
+        }
+    }
+
+    protected function addUnversionedButtons(DataObject $record, FieldList $actions)
```
Comment (flamerohr) [inline src/VersionedGridFieldItemRequest.php]: missing docblock

File: src/VersionedGridFieldItemRequest.php [hunk 5]
```diff
@@ -257,4 +228,91 @@ protected function getRecordStatus()
...
+            __CLASS__ . '.BUTTONAPPLYCHANGES',
+            'Apply changes'
+        ))->addExtraClass('btn-primary font-icon-save');
+
+        if ($warn) {
+            $actions->push(LiteralField::create(
+                'warning',
+                '<span class="btn actions-warning font-icon-info-circled">'
+                    ._t(
+                        __CLASS__ . '.WARNINGAPPLYCHANGES',
```
Comment (flamerohr) [inline src/VersionedGridFieldItemRequest.php]: I've changed the translation key to `PUBLISHITEMSWARNING`

**Comment by tractorcow** [comment]
Forgot to remove this obsolete config


```yaml
---
Name: versionedownership-gridfield
---
SilverStripe\Forms\GridField\GridFieldDetailForm_ItemRequest:
  extensions:
    RecursivePublishableHandler: SilverStripe\Versioned\RecursivePublishableHandler
```

**Pull request merged**